include/rocksdb/options.h

```diff
@@ -410,6 +410,8 @@ struct ColumnFamilyOptions : public AdvancedColumnFamilyOptions {
   // Create ColumnFamilyOptions from Options
   explicit ColumnFamilyOptions(const Options& options);
 
+  bool is_transient = false;
```

Comment:
Can you add comment to explain what the option is for? Maybe include that opening a DB will fail if user passes in a transient CF, and mention forward/backward compatibility.